source/http2_stream_manager.c

```diff
@@ -390,11 +394,119 @@ static void s_aws_http2_stream_manager_build_transaction_synced(struct aws_http2
     s_sm_log_stats_synced(stream_manager);
 }
 
+static void s_on_ping_complete(
+    struct aws_http_connection *http2_connection,
+    uint64_t round_trip_time_ns,
+    int error_code,
+    void *user_data) {
+
+    (void)http2_connection;
+    struct aws_h2_sm_connection *sm_connection = user_data;
+    if (error_code) {
+        return;
```

Comment:
don't we need to do ref-counting on `sm_connection` to be sure it stays alive until the ping_complete is received?

Reply:
yeah, you are right.